ncbi/src/main/java/org/reactome/release/NCBIEntry.java

```diff
@@ -0,0 +1,201 @@
+		while (result.hasNext()) {
+			Record record = result.next();
+			long uniprotDbId = record.get("rgp.dbId").asLong();
+			String uniprotDisplayName = record.get("rgp.displayName").asString();
+			String uniprotAccession = record.get("rgp.identifier").asString();
+			String ncbiGeneID = record.get("rds.identifier").asString();
+
+			UniProtReactomeEntry uniprot = UniProtReactomeEntry.get(uniprotDbId, uniprotAccession, uniprotDisplayName);
+			Set<String> ncbiGeneIDs = uniprotToNCBIGene.computeIfAbsent(uniprot, k -> new HashSet<>());
+			ncbiGeneIDs.add(ncbiGeneID);
+		}
+
+		List<NCBIEntry> ncbiEntries = new ArrayList<>();
+		for (UniProtReactomeEntry uniprot : uniprotToNCBIGene.keySet()) {
+			ncbiEntries.add(
+				new NCBIEntry(uniprot, uniprotToNCBIGene.get(uniprot))
+			);
+		}
+		Collections.sort(ncbiEntries);
+
+		logger.info("Finished generating UniProt accession to NCBI Gene mapping");
+
+		return ncbiEntries;
+	}
+
+	/**
+	 * Compares UniProt accession values of this object and parameter
+	 * @param o NCBIEntry object to compare
+	 * @return Value of String compare between this UniProt accession and the parameter's UniProt accession
+	 */
+	@Override
+	public int compareTo(@Nonnull NCBIEntry o) {
+		return this.getUniprotAccession().compareTo(o.getUniprotAccession());
+	}
+
+	/**
+	 * Checks equality based on object type and value of UniProt accession, UniProt display name, and NCBI Gene ids
+	 * @param o Object to check for equality with the calling NCBI Entry object.
+	 * @return <code>true</code> if the same object or an NCBI Entry object with the same UniProt accession,
+	 * UniProt display name, and NCBI Gene ids.  Returns <code>false</code> otherwise.
+	 */
+	@Override
+	public boolean equals(Object o) {
+		if (o == this) {
+			return true;
+		}
+
+		if (!(o instanceof NCBIEntry)) {
+			return false;
+		}
+
+		NCBIEntry other = (NCBIEntry) o;
+
+		return other.getUniprotAccession().equals(this.getUniprotAccession()) &&
+			   other.getUniprotDisplayName().equals(this.getUniprotDisplayName()) &&
+			   other.getNcbiGeneIds().equals(this.getNcbiGeneIds());
+	}
+
+	/**
+	 * Retrieves a hash code based on the object's set fields
+	 * @return Hash code of NCBI Entry object
+	 */
+	@Override
+	public int hashCode() {
+		return Objects.hash(getUniprotAccession(), getUniprotDisplayName(), getNcbiGeneIds());
+	}
+
+	/**
+	 * Retrieves a String representation of the defining data of the NCBI Entry object
+	 * @return String representation of NCBI Entry object
+	 */
+	@Override
+	public String toString() {
+		return this.getUniprotDisplayName() + " with NCBI Gene ids " + this.getNcbiGeneIds();
+	}
+
+	/**
+	 * Returns the XML String describing the relationship between the object's UniProt entry and an NCBI Gene
+	 * @param ncbiGene NCBI Gene identifier
+	 * @return XML String for NCBI Gene Entity Link
+	 */
+	public String getEntityLinkXML(String ncbiGene) {
+		return getLinkXML(
+			ncbiGene,
+			"&entity.base.url;",
```

Comment:
Does the string "&entity.base.url" actually get emitted into the output XML? I think the `&` might give you problems. You might need to encode it as `&amp;`.

Reply:
Yes, the string with the ampersand needs to be in the output file as part of the XML content.  I'm a little reluctant to change it as NCBI hasn't said it's a problem for them, but I can send them an e-mail to ask if the `&amp;` encoding would work for them so it is proper.

Reply:
Ok. Because I'm quite certain that an XML file containing `&entity.base.url` will fail basic XML validation (probably complain that `&entity.base.url` is not a valid entity, _or_ that it's missing a `;` at the end).

But if NCBI is already accepting files with this in them, I guess it's not a problem.

Reply:
Actually, I found this is okay because the headers NCBI specified to use are for defining special values of which entity.base.url and event.base.url are two.

https://www.w3schools.com/xml/xml_dtd_entities.asp

Reply:
Ah, so they declare their own entity types.